app/component/saltMachine/saltMachine.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const angular = require('angular');
+
+angular.module('saltMachineApp')
+.component('saltMachine', {
+  template: `<div>
+  <h1> run away from the salt machine </h1>
+  <console> </console>
+  <history> </history>
+  </div>`,
+  controllerAs: 'historyCtrl',
+  controller: ['playerService', function(playerService){
+    this.$onInit = () => {
+      this.player = playerService;
+    };
+  }],
+});
```

Comment:
**TODO FOR LOST POINTS**

You actually don't need lines 12-17 here. This is actually causing a bug where every time I hit the move button, I kept getting "Slug hit a wall" every time. I tracked it down to this module here *AND* your ``index.html.`` 

Basically, you were doing the same thing twice. You already have your ``console`` and ``history`` components at lines 09-10 displayed here, but they were also included duplicately in your ``index.html.`` (This may be my bad as I may have advised you to do that).

This somehow caused your bug where your history was only recording the "You hit a wall" part of your code instead of actually moving to locations. 

See my comments in ``index.html`` on what to fix there. 


See comments there on what to fix there.